### PR TITLE
Update quick tutorial with new problem variants

### LINKED ISSUE #486: Add SPD / backhaul example to the docs
I'm exploring PyVRP for pickup and delivery problems but couldn't find any example in the docs.
Can you please provide an example for such a scenario? thanks

### PR DESCRIPTION
**Pull request by N-Wouda**
This PR:

- [x] Closes #486.
- [x] Adds and passes relevant tests.
- [x] Has well-formatted code and documentation.

<details>

**Notes**:

- It is essential that you add a test when making code changes.
  This keeps the code coverage level up, and helps ensure the changes work as intended.
- When fixing a bug, you must add a test that would produce the bug in the master branch, and then show that it is fixed with the new code. 
- New code additions must be well formatted. Changes should pass the pre-commit workflow, which you can set up locally using [pre-commit](https://pre-commit.com/#intro). 
- Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: examples/quick_tutorial.ipynb [hunk 1]
```diff
@@ -7,10 +7,12 @@
    "source": [
     "# A quick tutorial\n",
     "\n",
-    "This notebook provides a brief tutorial to modelling vehicle routing problems with PyVRP, introducing some of its most important modelling features.\n",
-    "We will first solve a capacitated VRP, introducing the modelling interface and the most basic components.\n",
-    "We then solve a VRP with time windows, where we introduce the support PyVRP has for problems with time constraints.\n",
-    "Finally, we solve a multi-depot VRP with time windows, to showcase some additional features."
+    "This notebook provides a brief tutorial to modelling vehicle routing problems with PyVRP, introducing some of its most important modelling features:\n",
+    "- We first solve a capacitated VRP, introducing the modelling interface and the most basic components.\n",
```
Comment (leonlan) [inline examples/quick_tutorial.ipynb:11]: Need a new line here to render itemize

File: examples/quick_tutorial.ipynb [hunk 2]
```diff
@@ -432,11 +434,233 @@
...
+    "_, ax = plt.subplots(figsize=(8, 8))\n",
+    "plot_solution(res.best, m.data(), plot_clients=True, ax=ax)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Some clients are not visited in the figure above.\n",
+    "These clients are all too far from the depots and other clients for their prizes to be worth more than the additional travel distance the visit would incur.\n",
```
Comment (leonlan) [inline examples/quick_tutorial.ipynb:550]:
```suggestion
    "These clients are all too far from other locations to visit: their prizes are not worth the additional travel distance.\n",
```

Two suggestions: 
- the depots and other clients -> other locations
- (optional) some rephrasing to make the sentence shorter.

File: tests/plotting/baseline_images/test_plotting/plot_solution_multiple_depots.png
Comment (leonlan) [inline tests/plotting/baseline_images/test_plotting/plot_solution_multiple_depots.png:1]: This is a bit of an odd edge case where there are two routes, but the legend only shows `Route 2`.  It's super niche though so I feel we have to do anything with this.

```python
       # Coordinates of clients served by this route.
        if len(route) == 1 or plot_clients:
            ax.scatter(x, y, label=f"Route {idx}", zorder=3, s=75)
        ax.plot(x, y)
```

File: examples/quick_tutorial.ipynb (hunk 1 above)
Comment (N-Wouda) [reply examples/quick_tutorial.ipynb:11]: Every Markdown parser handles this slightly differently. The one I use locally is fine with this, but clearly whatever Sphinx uses is not. I'll update this to make it work with Sphinx.

**Review by leonlan — APPROVED** [review]

File: tests/plotting/baseline_images/test_plotting/plot_solution_multiple_depots.png
Comment (leonlan) [reply tests/plotting/baseline_images/test_plotting/plot_solution_multiple_depots.png:1]: I guess zo
